core/eth/reader.go

```diff
@@ -656,44 +659,159 @@ func (t *Reader) GetAllVersionedBlobParams(ctx context.Context) (map[uint8]*core
+		return nil, err
+	}
+
+	// since reservations are returned in the same order as the accountIDs, we can directly map them
+	for i, reservation := range reservations {
+		reservationsMap[accountIDs[i]] = &reservation
+	}
+
+	// filter out all zero-valued reservations
+	for accountID, reservation := range reservationsMap {
+		if isZeroValuedReservation(*reservation) {
+			delete(reservationsMap, accountID)
+		}
+	}
+
+	return reservationsMap, nil
 }
 
-func (t *Reader) GetActiveReservationByAccount(ctx context.Context, blockNumber uint32, accountID string) (core.ActiveReservation, error) {
-	// contract is not implemented yet
-	return core.ActiveReservation{}, nil
+func (t *Reader) GetActiveReservationByAccount(ctx context.Context, accountID gethcommon.Address) (*core.ActiveReservation, error) {
+	if t.bindings.PaymentVault == nil {
+		return nil, errors.New("payment vault not deployed")
+	}
+	reservation, err := t.bindings.PaymentVault.GetReservation(&bind.CallOpts{
+		Context: ctx,
+	}, accountID)
+	if err != nil {
+		return nil, err
+	}
+	if isZeroValuedReservation(reservation) {
+		return nil, errors.New("reservation does not exist for given account")
+	}
+	return &reservation, nil
 }
 
-func (t *Reader) GetOnDemandPayments(ctx context.Context, blockNumber uint32, accountIDs []string) (map[string]core.OnDemandPayment, error) {
-	// contract is not implemented yet
-	return map[string]core.OnDemandPayment{}, nil
+func (t *Reader) GetOnDemandPayments(ctx context.Context, accountIDs []gethcommon.Address) (map[gethcommon.Address]*core.OnDemandPayment, error) {
+	if t.bindings.PaymentVault == nil {
+		return nil, errors.New("payment vault not deployed")
+	}
+	paymentsMap := make(map[gethcommon.Address]*core.OnDemandPayment)
+	payments, err := t.bindings.PaymentVault.GetOnDemandAmounts(&bind.CallOpts{
+		Context: ctx,
+	}, accountIDs)
+	if err != nil {
+		return nil, err
+	}
+
+	// since payments are returned in the same order as the accountIDs, we can directly map them
+	for i, payment := range payments {
+		paymentsMap[accountIDs[i]] = &core.OnDemandPayment{
+			CumulativePayment: payment,
+		}
+	}
+
+	// filter out all zero-valued payments
+	for accountID, payment := range paymentsMap {
+		if payment.CumulativePayment.Cmp(big.NewInt(0)) == 0 {
+			delete(paymentsMap, accountID)
+		}
+	}
+
+	return paymentsMap, nil
 }
 
-func (t *Reader) GetOnDemandPaymentByAccount(ctx context.Context, blockNumber uint32, accountID string) (core.OnDemandPayment, error) {
-	// contract is not implemented yet
-	return core.OnDemandPayment{}, nil
+func (t *Reader) GetOnDemandPaymentByAccount(ctx context.Context, accountID gethcommon.Address) (*core.OnDemandPayment, error) {
+	if t.bindings.PaymentVault == nil {
+		return nil, errors.New("payment vault not deployed")
+	}
+	onDemandPayment, err := t.bindings.PaymentVault.GetOnDemandAmount(&bind.CallOpts{
+		Context: ctx,
+	}, accountID)
+	if err != nil {
+		return nil, err
+	}
+	if onDemandPayment == big.NewInt(0) {
```

Comment:
Has this been tested? This will result in false even if `onDemandPayment` has value of 0 because you're comparing bigint pointers. 
Use something like `onDemandPayment.Cmp(big.NewInt(0)) == 0` instead

Reply:
I don't have test coverage over this. Do we have anywhere that puts tests on contract calls?
(updated to use your suggestion)